### PR TITLE
[top/dv] Testplan updates

### PR DESCRIPTION
**Pull request by tjaychen**
- Enhance testplan descriptions due to reset domain change
- Move P3 tests to V3.

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
Added testplan description based on the reset domain change that is incoming.
Once this PR is merged, appropriate issues need to be created or re-opened.

**Review by msfschaffner — APPROVED** [review]

**Comment by tjaychen** [comment]
yeah i plan to file once this is approved.

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 1]
```diff
@@ -575,6 +575,7 @@
 
             - Program some CSRs / mem that are under life cycle reset tree and system reset tree.
             - Configure RV_DM to send NDM reset request to reset sytem reset tree.
+            - While NDM reset is ongoing, ensure the RV_DM debug module registers can still be accessed.
```
[the remark this answers is not in the document]
Comment (tjaychen) [reply hw/top_earlgrey/data/chip_testplan.hjson:578]: yeah i guess we didn't consider this in the original template..like a test "update".
I should probably go add that into the template.

[the remark this answers is not in the document]
Comment (sriyerg) [reply hw/top_earlgrey/data/chip_testplan.hjson:578]: Should this be a block level test?

[the remark this answers is not in the document]
Comment (tjaychen) [reply hw/top_earlgrey/data/chip_testplan.hjson:578]: ah this goes along with @jeoongp PR of reset domain change.
So this actually is a cross check between pinmux / rv_dm as a result of that change. So I think this one actually does need to be at top level.

**Review by jeoongp — APPROVED** [review]

**Review by jeoongp — APPROVED** [review]
LGTM

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 1 above)
[the remark this answers is not in the document]
Comment (sriyerg) [reply hw/top_earlgrey/data/chip_testplan.hjson:578]: SG.

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 2]
```diff
@@ -634,6 +636,16 @@
       stage: V2
       tests: ["chip_rv_dm_lc_disabled"]
     }
+    {
+      name: chip_rv_dm_access_after_hw_reset
```
[the remark this answers is not in the document]
Comment (sriyerg) [reply hw/top_earlgrey/data/chip_testplan.hjson:640]: Done in #15796.

**Pull request merged**